base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -201,30 +201,35 @@ macro_rules! fetch {
 /// let db_backend = MemoryDatabase::<HashDigest>::default();
 /// let validators = Validators::new(MockValidator::new(true), MockValidator::new(true));
 /// let db = MemoryDatabase::<HashDigest>::default();
-/// let mut db = BlockchainDatabase::new(db_backend, validators).unwrap();
+/// let mut db = BlockchainDatabase::new(db_backend).unwrap();
+/// db.add_validators(validators);
 /// // Do stuff with db
 /// ```
 pub struct BlockchainDatabase<T>
 where T: BlockchainBackend
 {
     metadata: Arc<RwLock<ChainMetadata>>,
     db: Arc<T>,
-    validators: Validators<T>,
+    validators: Option<Validators<T>>,
 }
 
 impl<T> BlockchainDatabase<T>
 where T: BlockchainBackend
 {
     /// Creates a new `BlockchainDatabase` using the provided backend.
-    pub fn new(db: T, validators: Validators<T>) -> Result<Self, ChainStorageError> {
+    pub fn new(db: T) -> Result<Self, ChainStorageError> {
         let metadata = Self::read_metadata(&db)?;
         Ok(BlockchainDatabase {
             metadata: Arc::new(RwLock::new(metadata)),
             db: Arc::new(db),
-            validators,
+            validators: None,
         })
     }
 
+    pub fn add_validators(&mut self, validators: Validators<T>) {
```

Comment:
We're setting the validators here, not adding them to a list
```suggestion
    pub fn set_validators(&mut self, validators: Validators<T>) {
```